### PR TITLE
[Regression testing] Simple linear temperature profile regression test

### PR DESCRIPTION
**Pull request by JBludau**
Currently for PMB only but already prepared for when we have thermomechanics in the LPS

Uses made up material properties as it compares to an analytical solution.

### CODE REVIEW

File: regression_test/tstCubeLinearTemperature.hpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        low_corner[1] + horizon, high_corner[1] - horizon,
+        low_corner[2] + horizon, high_corner[2] - horizon );
+    CabanaPD::Region<CabanaPD::RectangularPrism> full_cube(
+        low_corner[0], high_corner[0], low_corner[1], high_corner[1],
+        low_corner[2], high_corner[2] );
+
+    double edge_length_cube = high_corner[0] - low_corner[0] - 2 * horizon;
+
+    // Create BC last to ensure ghost particles are included.
+    auto u = solver.particles.sliceDisplacement();
```
Comment (streeve) [inline regression_test/tstCubeLinearTemperature.hpp:108]: Need to reslice x & temp here to match sizes

File: regression_test/tstCubeLinearTemperature.hpp [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        if ( !center_cube.inside( x, pid ) )
+        {
+            // uni-axial temperature profile
+            temp( pid ) =
+                x( pid, 0 ) / edge_length_cube * delta_temp * non_dim_time;
+        }
+    };
+    auto bc = CabanaPD::createBoundaryCondition(
+        linear_temp_func, exec_space{}, solver.particles, false, full_cube );
+
```
Comment (streeve) [inline regression_test/tstCubeLinearTemperature.hpp]:
```suggestion
    CabanaPD::BodyTerm bc(
        linear_temp_func, solver.particles.size(), false );
```
Since the intent is to include all particles

File: regression_test/tstCubeLinearTemperature.hpp (hunk 1 above)
Comment (JBludau) [reply regression_test/tstCubeLinearTemperature.hpp:108]: ah thanks. That makes sense ... but is pretty hard to spot

Comment (streeve) [reply regression_test/tstCubeLinearTemperature.hpp:108]: You're right, we need a runtime check like we have for the boundary conditions

**Comment by JBludau** [comment]
@pabloseleson this is ready. But we might want to think about which values we want to test. Right now it is just a 1deg temperature change to make it run fast.

Screenshot 2026-09-14 at 1 45 10 PM

**Comment by JBludau** [comment]
Temperature on the centerline of the cube along x-direction
Screenshot 2026-09-14 at 2 39 45 PM

File: regression_test/tstCubeLinearTemperature.hpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    CabanaPD::Solver solver( inputs, particles, force_model );
+
+    // ====================================================
+    //            Boundary condition
+    // ====================================================
+    CabanaPD::Region<CabanaPD::RectangularPrism> center_cube(
+        low_corner[0] + horizon, high_corner[0] - horizon,
+        low_corner[1] + horizon, high_corner[1] - horizon,
+        low_corner[2] + horizon, high_corner[2] - horizon );
+    CabanaPD::Region<CabanaPD::RectangularPrism> full_cube(
```
Comment (streeve) [inline regression_test/tstCubeLinearTemperature.hpp]: Can remove

File: regression_test/inputs/cube_linear_temperature.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "num_cells"              : {"value": [16, 16, 16]},
+    "system_size"            : {"value": [16e-4, 16e-4, 16e-4], "unit": "m"},
+    "density"                : {"value": 1, "unit": "kg/m^3"},
+    "elastic_modulus"        : {"value": 1e-9, "unit": "Pa"},
+    "fracture_energy"        : {"value": 3000, "unit": "J/m^2"},
+    "thermal_conductivity"    : {"value": 3, "unit": "W/(m.K)"},
+    "specific_heat_capacity"  : {"value": 1, "unit": "J/(kg.K)"},
+    "reference_temperature"   : {"value": 20.0, "unit": "oC"},
+    "horizon"                : {"value": 0.0003, "unit": "m"},
+    "yield_stress"           : {"value": 30e6, "unit": "Pa"},
```
Comment (streeve) [inline regression_test/inputs/cube_linear_temperature.json]:
```suggestion
    "thermal_conductivity"    : {"value": 3, "unit": "W/(m.K)"},
    "specific_heat_capacity"  : {"value": 1, "unit": "J/(kg.K)"},
    "reference_temperature"   : {"value": 20.0, "unit": "oC"},
    "horizon"                : {"value": 0.0003, "unit": "m"},
```

File: regression_test/tstCubeLinearTemperature.hpp
Comment (streeve) [inline regression_test/tstCubeLinearTemperature.hpp:1]: rename CubeLinearHeatTransfer?

**Review by streeve — APPROVED** [review]
Looks good - I would rather start here and improve later

File: regression_test/tstCubeLinearTemperature.hpp (hunk 3 above)
Comment (JBludau) [reply regression_test/tstCubeLinearTemperature.hpp]: oh yeah, I missed that we made a bc without a region.

Comment (JBludau) [reply regression_test/tstCubeLinearTemperature.hpp]: done in https://github.com/ORNL/CabanaPD/pull/313/commits/728af00d2caf2580883e005a5a85ac262249ebc2

File: regression_test/tstCubeLinearTemperature.hpp
Comment (JBludau) [reply regression_test/tstCubeLinearTemperature.hpp:1]: I was trying to name them by the boundary conditions, but am fine either way.
